src/mds/Server.cc

```diff
@@ -4452,6 +4452,26 @@ void Server::handle_client_openc(MDRequestRef& mdr)
     return;
   }
 
+  CF_MDS_RetryRequestFactory cf(mdcache, mdr, false);
+  int r = mdcache->path_traverse(mdr, cf, req->get_filepath(), 0, NULL);
+  if (r > 0) return;
+  if (r == 0) {
+    // it existed.
+    handle_client_open(mdr);
+    return;
+  }
+  if (r < 0 && r != -CEPHFS_ENOENT) {
+    if (r == -CEPHFS_ESTALE) {
```

Comment:
How about

```
if (r < 0) {
    if (r != -CEPHFS_ENOENT && r == -CEPHFS_ESTALE) {
........
```

Reason being it looks clean and aligns with above conditions, i.e. `if (r > 0)` then `if (r == 0)` and then `if (r < 0)`, but it's just a suggestion.